iron-overlay-behavior.html

```diff
@@ -549,23 +565,98 @@
     },
 
     /**
-     * Executes a callback on the next animation frame, overriding any previous
-     * callback awaiting for the next animation frame. e.g.
-     * `__onNextAnimationFrame(callback1) && __onNextAnimationFrame(callback2)`;
-     * `callback1` will never be invoked.
-     * @param {!Function} callback Its `this` parameter is the overlay itself.
+     * Delays the execution of a callback to the next animation frame.
+     * @param {!string} callbackName
      * @private
      */
-    __onNextAnimationFrame: function(callback) {
-      if (this.__raf) {
-        window.cancelAnimationFrame(this.__raf);
-      }
-      var self = this;
-      this.__raf = window.requestAnimationFrame(function nextAnimationFrame() {
-        self.__raf = null;
-        callback.call(self);
+    __deraf: function(callbackName) {
+      var rafs = this.__rafs;
+      rafs[callbackName] && cancelAnimationFrame(rafs[callbackName]);
+      var callback = this[callbackName].bind(this);
+      rafs[callbackName] = requestAnimationFrame(function nextAnimationFrame() {
+        rafs[callbackName] = null;
+        callback();
       });
-    }
+    },
+
+    /**
+     * @param {boolean} isAttached
+     * @param {boolean} opened
+     * @param {boolean} withBackdrop
+     * @param {boolean} allowOutsideScroll
+     * @private
+     */
+    __lockOutsideScroll: function(isAttached, opened, withBackdrop, allowOutsideScroll) {
+      if (isAttached && opened && (withBackdrop || !allowOutsideScroll)) {
+        Polymer.IronScrollManager.pushScrollLock(this);
+      } else {
+        Polymer.IronScrollManager.removeScrollLock(this);
+      }
+    },
+
+    /**
+     * @param {boolean} isAttached
+     * @param {boolean} opened
+     * @private
+     */
+    __updateScrollListener: function(isAttached, opened) {
+      if (isAttached && opened) {
```

Comment:
Can't you just add `allowOutsideScroll` (and depending on the previous comment `withBackdrop`) as another listened variable? I don't think it would add too much overhead.